src/reducers/platform.js

```diff
@@ -19,45 +20,52 @@ import {
 	CACHE_SUCCESS
 } from '../actions/cacheActionCreators';
 
+type ApiState = {
+	inFlight: Array<string>,
+	fail?: boolean,
+	[string]: QueryResponse,
+};
+
+export const DEFAULT_API_STATE: ApiState = { inFlight: [] };
 export const DEFAULT_APP_STATE = { isFetching: false };
 
-export const responseToState = response => {
-	const { ref, ...data } = response;
-	return { [ref]: data };
-};
+export const responseToState = (response: QueryResponse): { [string]: QueryResponse } =>
+	({ [response.ref]: response });
 
-/**
+/*
  * The primary reducer for data provided by the API
- * `state.api` sub-tree
- *
- * @param {Object} state
- * @param {ReduxAction} action
- * @return {Object}
  */
-export function api(state=DEFAULT_APP_STATE, action={}) {
+export function api(state: ApiState = DEFAULT_API_STATE, action: FluxStandardAction): ApiState {
 	switch (action.type) {
-	case API_REQ:
+	case API_REQ: {
+		const requestRefs = (action.payload || []).map(({ ref }) => ref);
+		const inFlight = state.inFlight
+			.filter(ref => !requestRefs.includes[ref]) // clean out current duplicates
+			.concat(requestRefs); // add new requested refs
```

Comment:
Can you explain again why it makes more sense to keep track of requests in an array versus an object?  I can why semantically it makes sense because its a list of objects. But it seems like it would simplify this step and every other step?